packages/@aws-cdk/aws-iotevents/lib/input.ts

```diff
@@ -8,11 +9,55 @@ import { CfnInput } from './iotevents.generated';
 export interface IInput extends IResource {
   /**
    * The name of the input
+   *
    * @attribute
    */
   readonly inputName: string;
+
+  /**
+   * The ARN of the input
+   *
+   * @attribute
+   */
+  readonly inputArn: string;
+
+  /**
+   * Grant the putting message permission to the given IAM principal (Role/Group/User).
+   *
+   * @param grantee the principal
+   */
+  grantPutMessage(grantee: iam.IGrantable): iam.Grant
+
+  /**
+   * Grant the indicated permissions on this input to the given IAM principal (Role/Group/User).
+   *
+   * @param grantee the principal
+   * @param actions the set of actions to allow (i.e. "iotevents:BatchPutMessage")
+   */
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant
 }
 
+
+abstract class InputBase extends Resource implements IInput {
+  public abstract readonly inputName: string;
+
+  public abstract readonly inputArn: string;
+
+  public grantPutMessage(grantee: iam.IGrantable) {
+    return this.grant(grantee, 'iotevents:BatchPutMessage');
+  }
+
+  public grant(grantee: iam.IGrantable, ...actions: string[]) {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [this.inputArn],
+      scope: this,
```

Comment:
Hmm. Why are we passing this argument here? I don't think it's required?

Reply:
Oh. I copied from [DynamoDB Table grant](https://github.com/yamatatsu/aws-cdk/blob/af4114e00ffd979af4f5b0c4204fcde343163126/packages/@aws-cdk/aws-dynamodb/lib/table.ts#L634). I removed and confirm the test is passed.